packages/flutter/lib/src/widgets/image_icon.dart

```diff
@@ -48,12 +49,18 @@ class ImageIcon extends StatelessWidget {
   /// The color to use when drawing the icon.
   ///
   /// Defaults to the current [IconTheme] color, if any. If there is
-  /// no [IconTheme], then it defaults to not recolorizing the image.
+  /// no [IconTheme], then it defaults to [IconThemeData.fallback].
   ///
   /// The image will additionally be adjusted by the opacity of the current
   /// [IconTheme], if any.
   final Color color;
 
+  /// Flag to decide if ImageIcon should recolor the image or not.
+  ///
+  /// Defaults to true, if set to false ImageIcon will ignore any color
+  /// attribute including [IconTheme] color too.
+  final bool shouldRecolor;
```

Comment:
This name isn't immediately clear (should recolor what? to what?).

How about `honorIconThemeColor` or `ignoreIconThemeColor` or some such?